src/shared/verify.js

```diff
@@ -0,0 +1,69 @@
+const { getData, validateSchema } = require("@govtechsg/open-attestation");
+const {
+  verificationBuilder,
+  openAttestationVerifiers
+} = require("@govtechsg/oa-verify");
+const { registryVerifier } = require("@govtechsg/opencerts-verify");
+
+const config = require("./config");
+
+const IS_MAINNET =
+  config.network === "mainnet" || config.network === "homestead";
+
+function getNetworkName(document) {
+  const data = getData(document);
+
+  if (IS_MAINNET && data.network) {
+    /* Production Network Whitelist */
+    switch (data.network?.chainId) {
+      case "137":
+        return "matic";
+      default:
+        break;
+    }
+  } else {
+    /* Non-production Network Whitelist */
+    switch (data.network?.chainId) {
+      case "80002":
+        // return "amoy";
+        // FIXME: Setting "amoy" will fail as it's an unsupported network in Ethers v5.7.2
+        // Create a custom provider and specify { chainId: 80002, name: "amoy" }
+        // https://github.com/OpenCerts/opencerts-website/blob/1de65c66795ec2f416d6c829259e9f9ab1e49e45/src/sagas/certificate.ts#L123
+        console.error(`"amoy" is not supported on Ethers v5.7.2 yet`);
+        break;
+      default:
+        break;
+    }
+  }
+
+  // A network is specified in the certificate but not in the above whitelist
+  if (data.network) {
+    console.warn(
+      `"${JSON.stringify(
+        data.network
+      )}" is not a whitelisted network. Reverting back to "${config.network}".`
+    );
+  }
+
+  return config.network;
+}
+
+/**
+ * A wrapper of verify to auto-switch between Ethereum and Polygon
+ */
+export const verify = (document) => {
+  if (!validateSchema(document)) {
+    // Following current behaviour of from "@govtechsg/opencerts-verify"
+    // E.g. await verify({ "bad": "document" }) // returns undefined
```

Comment:
weird is the return type for `verify` wrong? it should not return an `undefined` based on the type haha

Reply:
I did some further digging and realised it wasn't `verify()` that was causing the issue I was facing originally.

 `registryVerifier` will not handle a bad document like `{ foo: "bar" }` correctly and throw this error:
```
TypeError: Cannot read properties of undefined (reading 'test')
```

Reply:
i see!

Reply:
I've created a temporary branch (`temp/type-error`) to showcase the error: https://github.com/OpenCerts/opencerts-functions/commit/9f6bdd25b2268896f17c1f421f51f3a5dac440c3

In the meantime for this PR, will revert to using `opencerts-verify` so we don't have to manually build a `verify` instance.